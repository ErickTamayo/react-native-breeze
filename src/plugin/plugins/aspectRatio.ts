import { PluginFunction, PluginPattern } from "./types";

export type PluginGroups = {
  key: string;
};

export const pattern: PluginPattern = ({ keys }) => {
  return new RegExp(`^aspect-ratio-(?<key>${keys("aspectRatio")})$`);
};

export const plugin: PluginFunction<PluginGroups> = ({ groups, theme }) => {
  const { key } = groups;
  const value = theme(["aspectRatio", key]);
  return { aspectRatio: value };
};
