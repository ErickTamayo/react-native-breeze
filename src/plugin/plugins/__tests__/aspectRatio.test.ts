import {
  generateInput,
  shouldEvaluateTheCorrectPatternTest,
  shouldMatchOutputSnapshot,
} from "../../helpers/tests";
import { pattern, plugin } from "../aspectRatio";

describe("aspectRatio", () => {
  // prettier-ignore
  const shouldMatch = generateInput("aspect-ratio", ["1"]);

  shouldEvaluateTheCorrectPatternTest(pattern, {
    shouldMatch,
    shouldNotMatch: ["unknown", "aspect-ratio-unknown"],
  });

  shouldMatchOutputSnapshot(pattern, plugin, shouldMatch);
});
