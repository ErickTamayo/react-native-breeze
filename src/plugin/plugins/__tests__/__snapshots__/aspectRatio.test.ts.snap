// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`aspectRatio Should parse (aspect-ratio-1) correctly 1`] = `
Object {
  "aspectRatio": 1,
}
`;
